1115-i386-txt-Initialize-TPM-1.2-event-log-in-TXT-heap.patch

```diff
@@ -0,0 +1,163 @@
+From d9ba5e9769fb55b246bf76b3d812b85497610325 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Micha=C5=82=20=C5=BBygowski?= <michal.zygowski@3mdeb.com>
+Date: Wed, 31 Aug 2022 14:37:49 +0200
+Subject: [PATCH 1115/1118] i386/txt: Initialize TPM 1.2 event log in TXT heap
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>
+Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>
+---
+ grub-core/loader/i386/txt/txt.c | 61 +++++++++++++++++++++++++++++++--
+ include/grub/i386/txt.h         | 29 ++++++++++++++++
+ 2 files changed, 88 insertions(+), 2 deletions(-)
+
+diff --git a/grub-core/loader/i386/txt/txt.c b/grub-core/loader/i386/txt/txt.c
+index 201167ce9a44..f3afb6ad1d92 100644
+--- a/grub-core/loader/i386/txt/txt.c
++++ b/grub-core/loader/i386/txt/txt.c
+@@ -501,6 +501,24 @@ set_mtrrs_for_acmod (struct grub_txt_acm_header *hdr)
+   return GRUB_ERR_NONE;
+ }
+ 
++static void init_tpm12_event_log(struct grub_slaunch_params *slparams)
++{
++  struct event_log_container *elog;
++  elog = (struct event_log_container *)(grub_addr_t)slparams->tpm_evt_log_base;
```

Comment:
`slparams->tpm_evt_log_base` was set as `get_physical_target_address()` so it isn't safe to use as a pointer.